src/widget/wtrackmenu.cpp

```diff
@@ -616,56 +601,107 @@ void WTrackMenu::setupActions() {
     }
 }
 
-void WTrackMenu::setTrackIds(TrackIdList trackIdList) {
+void WTrackMenu::loadTracks(TrackIdList trackIdList) {
     // Clean all forms of track store
     clearTrackSelection();
 
-    // Store the track pointers at each initialization of track ids.
-    for (const auto trackId : trackIdList) {
-        TrackPointer trackPointer = m_pTrackCollectionManager->internalCollection()->getTrackById(trackId);
-        if (trackPointer) {
-            m_pTrackPointerList.push_back(trackPointer);
-            m_pTrackIdList.push_back(trackId);
+    TrackPointerList trackPointers;
+    trackPointers.reserve(trackIdList.size());
+    for (const auto& trackId : trackIdList) {
+        TrackPointer pTrack = m_pTrackCollectionManager->internalCollection()->getTrackById(trackId);
+        if (!pTrack) {
+            return;
+        }
+        trackPointers.push_back(pTrack);
+    }
+    if (trackPointers.empty()) {
+        return;
+    }
+
+    m_pTrackPointerList = trackPointers;
```

Comment:
Would be the same as `m_pTrackPointerList` has already been cleared. Assigning to it an empty list would make no difference.

Reply:
You're right, but I think it would make the code easier to understand to move `m_pTrackPointerList = trackPointers;` above 
```
    if (trackPointers.empty()) {
        return;
    }
```